model-providers/openai/openai-common/runtime/src/main/java/io/quarkiverse/langchain4j/openai/OpenAiRestApi.java

```diff
@@ -179,6 +182,32 @@ public boolean test(SseEvent<String> event) {
         }
     }
 
+    interface AuthProvider {
+        String getAuthorization(Input input);
+
+        record Input(
```

Comment:
This should be an interface as it is not something created by users, but instead by the framework